cantools/database/can/c_source.py

```diff
@@ -1523,11 +1530,14 @@ def _generate_definitions(database_name, messages, floating_point_numbers, use_f
 
                 signal_definitions.append(signal_definition)
 
-            signals_init_body += init_signal_body_template.format(type_name=signal.type_name,
-                                                                  signal_initial=signal.initial,
-                                                                  signal_name=signal.snake_name,
-                                                                  signal_data_length=int(signal.type_length / 8)
-                                                                  ) if signal.initial is not None else ""
+            if add_initializers and signal.initial:
+                init_value = float(signal.initial) * signal.scale + signal.offset
```

Comment:
```suggestion
                init_value = signal.initial
```

since #536 `signal.initial`  uses physical values for all supported file formats to make it consistent with `signal.{minimum,maximum}`. (before #536 it depended on the file format loader.)

Reply:
also, it seems like `INIT_SIGNAL_BODY_TEMPLATE_FMT` expects the internal representation of the value, i.e.,
```suggestion
                init_value = signal.initial_raw
```

Reply:
Thank you for pointing it out, my mistake.
I think exactly about assigning physical initial value in C code, not raw